### PR TITLE
improve settings ui performance

### PR DESCRIPTION
**Pull request by slaurent22**
- do not preload the full list for each split's combobox.
- add an edit button to toggle into the full selectible data source
- cache available splits (both sortings) to avoid duplicating the work
- leave files that were auto-generated by vs during the ui editing process

### CODE REVIEW

**Comment by cerpintext** [comment]
would it make sense to limit dragging of items to only those that are editable, and if so maybe mark an item for editing when it's dragged? 🤔

**Comment by slaurent22** [comment]
> would it make sense to limit dragging of items to only those that are editable, and if so maybe mark an item for editing when it's dragged? 🤔 

I personally like the flow of having the dragging be separate from the editing. Anecdotally, it seems re-ordering the existing autosplits is more common anyway

**Comment by cerpintext** [comment]
do you think it could be possible to make it possible to destinguish any splits you've moved from other splits? like make them less greyed out or something, just becaues i noticed i 100% lost track when testing with just moving one randomly :')

**Comment by slaurent22** [comment]
> do you think it could be possible to make it possible to destinguish any splits you've moved from other splits? like make them less greyed out or something, just becaues i noticed i 100% lost track when testing with just moving one randomly :')

I'll look into it :ax2uHappy:

**Comment by slaurent22** [comment]
I played around with modifying styling and it didn't look great. I ended up using your idea of enabling/marking editable when dragged

File: HollowKnightSettings.cs [hunk 1]
```diff
@@ -72,6 +81,48 @@ public partial class HollowKnightSettings : UserControl {
...
+                    HollowKnightSplitSettings setting = (HollowKnightSplitSettings)((Button)sender).Parent;
+                    int index = setting.Parent.Controls.GetChildIndex(setting);
+                    addSplitAtIndex(index);
+                }
+            }
+        }
+        private void btnAddBelow_Click(object sender, EventArgs e) {
+            for (int i = flowMain.Controls.Count - 1; i > 0; i--) {
+                if (flowMain.Controls[i].Contains((Control)sender)) {
+                    HollowKnightSplitSettings setting = (HollowKnightSplitSettings)((Button)sender).Parent;
```
Comment (slaurent22) [inline HollowKnightSettings.cs:109]: this code is repeated. if I had more energy I would look into abstracting it, but /shrug

Comment (cerpintext) [reply HollowKnightSettings.cs:109]: its 3 lines, abstracting that if its only used <5 times its a waste of effort innit
i'll check out all the thingies tomorrow when im awaketh once more and as an excuse to practice 😎

**Pull request closed without merging**